File: clients/ruby/models/instrument.rb
class Instrument
  attr_accessor :symbol, :root_symbol, :state, :typ, :listing, :front, :expiry, :relist_interval, :inverse_leg, :sell_leg, :buy_leg, :underlying, :quote_currency, :underlying_symbol, :reference, :reference_symbol, :tick_size, :multiplier, :settl_currency, :underlying_to_settle_multiplier, :quote_to_settle_multiplier, :is_quanto, :is_inverse, :init_margin, :maint_margin, :limit, :maker_fee, :taker_fee, :insurance_fee, :hedger_maker_fee, :hedger_taker_fee, :hedger_insurance_fee, :opening_timestamp, :closing_timestamp, :session_interval, :prev_close_price, :limit_down_price, :limit_up_price, :prev_total_volume, :total_volume, :volume, :volume24h, :prev_total_turnover, :total_turnover, :turnover, :turnover24h, :vwap, :high_price, :low_price, :last_price, :last_tick_direction, :last_change_pcnt, :bid_price, :mid_price, :ask_price, :open_interest, :open_value, :settled_price, :timestamp

  # :internal => :external
  def self.attribute_map
    {
      :symbol => :symbol,
      :root_symbol => :rootSymbol,
      :state => :state,
      :typ => :typ,
      :listing => :listing,
      :front => :front,
      :expiry => :expiry,
      :relist_interval => :relistInterval,
      :inverse_leg => :inverseLeg,
      :sell_leg => :sellLeg,
      :buy_leg => :buyLeg,
      :underlying => :underlying,
      :quote_currency => :quoteCurrency,
      :underlying_symbol => :underlyingSymbol,
      :reference => :reference,
      :reference_symbol => :referenceSymbol,
      :tick_size => :tickSize,
      :multiplier => :multiplier,
      :settl_currency => :settlCurrency,
      :underlying_to_settle_multiplier => :underlyingToSettleMultiplier,
      :quote_to_settle_multiplier => :quoteToSettleMultiplier,
      :is_quanto => :isQuanto,
      :is_inverse => :isInverse,
      :init_margin => :initMargin,
      :maint_margin => :maintMargin,
      :limit => :limit,
      :maker_fee => :makerFee,
      :taker_fee => :takerFee,
      :insurance_fee => :insuranceFee,
      :hedger_maker_fee => :hedgerMakerFee,
      :hedger_taker_fee => :hedgerTakerFee,
      :hedger_insurance_fee => :hedgerInsuranceFee,
      :opening_timestamp => :openingTimestamp,
      :closing_timestamp => :closingTimestamp,
      :session_interval => :sessionInterval,
      :prev_close_price => :prevClosePrice,
      :limit_down_price => :limitDownPrice,
      :limit_up_price => :limitUpPrice,
      :prev_total_volume => :prevTotalVolume,
      :total_volume => :totalVolume,
      :volume => :volume,
      :volume24h => :volume24h,
      :prev_total_turnover => :prevTotalTurnover,
      :total_turnover => :totalTurnover,
      :turnover => :turnover,
      :turnover24h => :turnover24h,
      :vwap => :vwap,
      :high_price => :highPrice,
      :low_price => :lowPrice,
      :last_price => :lastPrice,
      :last_tick_direction => :lastTickDirection,
      :last_change_pcnt => :lastChangePcnt,
      :bid_price => :bidPrice,
      :mid_price => :midPrice,
      :ask_price => :askPrice,
      :open_interest => :openInterest,
      :open_value => :openValue,
      :settled_price => :settledPrice,
      :timestamp => :timestamp

    }
  end

  def initialize(attributes = {})
    return if attributes.empty?
    # Morph attribute keys into undescored rubyish style
    if self.class.attribute_map[:"symbol"]
      @symbol = attributes["symbol"]
    end
    if self.class.attribute_map[:"root_symbol"]
      @root_symbol = attributes["rootSymbol"]
    end
    if self.class.attribute_map[:"state"]
      @state = attributes["state"]
    end
    if self.class.attribute_map[:"typ"]
      @typ = attributes["typ"]
    end
    if self.class.attribute_map[:"listing"]
      @listing = attributes["listing"]
    end
    if self.class.attribute_map[:"front"]
      @front = attributes["front"]
    end
    if self.class.attribute_map[:"expiry"]
      @expiry = attributes["expiry"]
    end
    if self.class.attribute_map[:"relist_interval"]
      @relist_interval = attributes["relistInterval"]
    end
    if self.class.attribute_map[:"inverse_leg"]
      @inverse_leg = attributes["inverseLeg"]
    end
    if self.class.attribute_map[:"sell_leg"]
      @sell_leg = attributes["sellLeg"]
    end
    if self.class.attribute_map[:"buy_leg"]
      @buy_leg = attributes["buyLeg"]
    end
    if self.class.attribute_map[:"underlying"]
      @underlying = attributes["underlying"]
    end
    if self.class.attribute_map[:"quote_currency"]
      @quote_currency = attributes["quoteCurrency"]
    end
    if self.class.attribute_map[:"underlying_symbol"]
      @underlying_symbol = attributes["underlyingSymbol"]
    end
    if self.class.attribute_map[:"reference"]
      @reference = attributes["reference"]
    end
    if self.class.attribute_map[:"reference_symbol"]
      @reference_symbol = attributes["referenceSymbol"]
    end
    if self.class.attribute_map[:"tick_size"]
      @tick_size = attributes["tickSize"]
    end
    if self.class.attribute_map[:"multiplier"]
      @multiplier = attributes["multiplier"]
    end
    if self.class.attribute_map[:"settl_currency"]
      @settl_currency = attributes["settlCurrency"]
    end
    if self.class.attribute_map[:"underlying_to_settle_multiplier"]
      @underlying_to_settle_multiplier = attributes["underlyingToSettleMultiplier"]
    end
    if self.class.attribute_map[:"quote_to_settle_multiplier"]
      @quote_to_settle_multiplier = attributes["quoteToSettleMultiplier"]
    end
    if self.class.attribute_map[:"is_quanto"]
      @is_quanto = attributes["isQuanto"]
    end
    if self.class.attribute_map[:"is_inverse"]
      @is_inverse = attributes["isInverse"]
    end
    if self.class.attribute_map[:"init_margin"]
      @init_margin = attributes["initMargin"]
    end
    if self.class.attribute_map[:"maint_margin"]
      @maint_margin = attributes["maintMargin"]
    end
    if self.class.attribute_map[:"limit"]
      @limit = attributes["limit"]
    end
    if self.class.attribute_map[:"maker_fee"]
      @maker_fee = attributes["makerFee"]
    end
    if self.class.attribute_map[:"taker_fee"]
      @taker_fee = attributes["takerFee"]
    end
    if self.class.attribute_map[:"insurance_fee"]
      @insurance_fee = attributes["insuranceFee"]
    end
    if self.class.attribute_map[:"hedger_maker_fee"]
      @hedger_maker_fee = attributes["hedgerMakerFee"]
    end
    if self.class.attribute_map[:"hedger_taker_fee"]
      @hedger_taker_fee = attributes["hedgerTakerFee"]
    end
    if self.class.attribute_map[:"hedger_insurance_fee"]
      @hedger_insurance_fee = attributes["hedgerInsuranceFee"]
    end
    if self.class.attribute_map[:"opening_timestamp"]
      @opening_timestamp = attributes["openingTimestamp"]
    end
    if self.class.attribute_map[:"closing_timestamp"]
      @closing_timestamp = attributes["closingTimestamp"]
    end
    if self.class.attribute_map[:"session_interval"]
      @session_interval = attributes["sessionInterval"]
    end
    if self.class.attribute_map[:"prev_close_price"]
      @prev_close_price = attributes["prevClosePrice"]
    end
    if self.class.attribute_map[:"limit_down_price"]
      @limit_down_price = attributes["limitDownPrice"]
    end
    if self.class.attribute_map[:"limit_up_price"]
      @limit_up_price = attributes["limitUpPrice"]
    end
    if self.class.attribute_map[:"prev_total_volume"]
      @prev_total_volume = attributes["prevTotalVolume"]
    end
    if self.class.attribute_map[:"total_volume"]
      @total_volume = attributes["totalVolume"]
    end
    if self.class.attribute_map[:"volume"]
      @volume = attributes["volume"]
    end
    if self.class.attribute_map[:"volume24h"]
      @volume24h = attributes["volume24h"]
    end
    if self.class.attribute_map[:"prev_total_turnover"]
      @prev_total_turnover = attributes["prevTotalTurnover"]
    end
    if self.class.attribute_map[:"total_turnover"]
      @total_turnover = attributes["totalTurnover"]
    end
    if self.class.attribute_map[:"turnover"]
      @turnover = attributes["turnover"]
    end
    if self.class.attribute_map[:"turnover24h"]
      @turnover24h = attributes["turnover24h"]
    end
    if self.class.attribute_map[:"vwap"]
      @vwap = attributes["vwap"]
    end
    if self.class.attribute_map[:"high_price"]
      @high_price = attributes["highPrice"]
    end
    if self.class.attribute_map[:"low_price"]
      @low_price = attributes["lowPrice"]
    end
    if self.class.attribute_map[:"last_price"]
      @last_price = attributes["lastPrice"]
    end
    if self.class.attribute_map[:"last_tick_direction"]
      @last_tick_direction = attributes["lastTickDirection"]
    end
    if self.class.attribute_map[:"last_change_pcnt"]
      @last_change_pcnt = attributes["lastChangePcnt"]
    end
    if self.class.attribute_map[:"bid_price"]
      @bid_price = attributes["bidPrice"]
    end
    if self.class.attribute_map[:"mid_price"]
      @mid_price = attributes["midPrice"]
    end
    if self.class.attribute_map[:"ask_price"]
      @ask_price = attributes["askPrice"]
    end
    if self.class.attribute_map[:"open_interest"]
      @open_interest = attributes["openInterest"]
    end
    if self.class.attribute_map[:"open_value"]
      @open_value = attributes["openValue"]
    end
    if self.class.attribute_map[:"settled_price"]
      @settled_price = attributes["settledPrice"]
    end
    if self.class.attribute_map[:"timestamp"]
      @timestamp = attributes["timestamp"]
    end
    

  end

  def to_body
    body = {}
    self.class.attribute_map.each_pair do |key, value|
      body[value] = self.send(key) unless self.send(key).nil?
    end
    body
  end
end

File: clients/dart/lib/model/instrument.dart
part of api;


@Entity()
class Instrument {
  
  @Property(name: 'symbol')
  String symbol = null;
  

  @Property(name: 'rootSymbol')
  String rootSymbol = null;
  

  @Property(name: 'state')
  String state = null;
  

  @Property(name: 'typ')
  String typ = null;
  

  @Property(name: 'listing')
  DateTime listing = null;
  

  @Property(name: 'front')
  DateTime front = null;
  

  @Property(name: 'expiry')
  DateTime expiry = null;
  

  @Property(name: 'settle')
  DateTime settle = null;
  

  @Property(name: 'relistInterval')
  DateTime relistInterval = null;
  

  @Property(name: 'inverseLeg')
  String inverseLeg = null;
  

  @Property(name: 'sellLeg')
  String sellLeg = null;
  

  @Property(name: 'buyLeg')
  String buyLeg = null;
  

  @Property(name: 'positionCurrency')
  String positionCurrency = null;
  

  @Property(name: 'underlying')
  String underlying = null;
  

  @Property(name: 'quoteCurrency')
  String quoteCurrency = null;
  

  @Property(name: 'underlyingSymbol')
  String underlyingSymbol = null;
  

  @Property(name: 'reference')
  String reference = null;
  

  @Property(name: 'referenceSymbol')
  String referenceSymbol = null;
  

  @Property(name: 'calcInterval')
  DateTime calcInterval = null;
  

  @Property(name: 'publishInterval')
  DateTime publishInterval = null;
  

  @Property(name: 'publishTime')
  DateTime publishTime = null;
  

  @Property(name: 'maxOrderQty')
  Number maxOrderQty = null;
  

  @Property(name: 'maxPrice')
  double maxPrice = null;
  

  @Property(name: 'lotSize')
  Number lotSize = null;
  

  @Property(name: 'tickSize')
  double tickSize = null;
  

  @Property(name: 'multiplier')
  Number multiplier = null;
  

  @Property(name: 'settlCurrency')
  String settlCurrency = null;
  

  @Property(name: 'underlyingToPositionMultiplier')
  Number underlyingToPositionMultiplier = null;
  

  @Property(name: 'underlyingToSettleMultiplier')
  Number underlyingToSettleMultiplier = null;
  

  @Property(name: 'quoteToSettleMultiplier')
  Number quoteToSettleMultiplier = null;
  

  @Property(name: 'isQuanto')
  bool isQuanto = null;
  

  @Property(name: 'isInverse')
  bool isInverse = null;
  

  @Property(name: 'initMargin')
  double initMargin = null;
  

  @Property(name: 'maintMargin')
  double maintMargin = null;
  

  @Property(name: 'riskLimit')
  Number riskLimit = null;
  

  @Property(name: 'riskStep')
  Number riskStep = null;
  

  @Property(name: 'limit')
  double limit = null;
  

  @Property(name: 'capped')
  bool capped = null;
  

  @Property(name: 'taxed')
  bool taxed = null;
  

  @Property(name: 'deleverage')
  bool deleverage = null;
  

  @Property(name: 'makerFee')
  double makerFee = null;
  

  @Property(name: 'takerFee')
  double takerFee = null;
  

  @Property(name: 'settlementFee')
  double settlementFee = null;
  

  @Property(name: 'insuranceFee')
  double insuranceFee = null;
  

  @Property(name: 'fundingBaseSymbol')
  String fundingBaseSymbol = null;
  

  @Property(name: 'fundingQuoteSymbol')
  String fundingQuoteSymbol = null;
  

  @Property(name: 'fundingPremiumSymbol')
  String fundingPremiumSymbol = null;
  

  @Property(name: 'fundingTimestamp')
  DateTime fundingTimestamp = null;
  

  @Property(name: 'fundingInterval')
  DateTime fundingInterval = null;
  

  @Property(name: 'fundingRate')
  double fundingRate = null;
  

  @Property(name: 'indicativeFundingRate')
  double indicativeFundingRate = null;
  

  @Property(name: 'rebalanceTimestamp')
  DateTime rebalanceTimestamp = null;
  

  @Property(name: 'rebalanceInterval')
  DateTime rebalanceInterval = null;
  

  @Property(name: 'openingTimestamp')
  DateTime openingTimestamp = null;
  

  @Property(name: 'closingTimestamp')
  DateTime closingTimestamp = null;
  

  @Property(name: 'sessionInterval')
  DateTime sessionInterval = null;
  

  @Property(name: 'prevClosePrice')
  double prevClosePrice = null;
  

  @Property(name: 'limitDownPrice')
  double limitDownPrice = null;
  

  @Property(name: 'limitUpPrice')
  double limitUpPrice = null;
  

  @Property(name: 'bankruptLimitDownPrice')
  double bankruptLimitDownPrice = null;
  

  @Property(name: 'bankruptLimitUpPrice')
  double bankruptLimitUpPrice = null;
  

  @Property(name: 'prevTotalVolume')
  Number prevTotalVolume = null;
  

  @Property(name: 'totalVolume')
  Number totalVolume = null;
  

  @Property(name: 'volume')
  Number volume = null;
  

  @Property(name: 'volume24h')
  Number volume24h = null;
  

  @Property(name: 'prevTotalTurnover')
  Number prevTotalTurnover = null;
  

  @Property(name: 'totalTurnover')
  Number totalTurnover = null;
  

  @Property(name: 'turnover')
  Number turnover = null;
  

  @Property(name: 'turnover24h')
  Number turnover24h = null;
  

  @Property(name: 'prevPrice24h')
  double prevPrice24h = null;
  

  @Property(name: 'vwap')
  double vwap = null;
  

  @Property(name: 'highPrice')
  double highPrice = null;
  

  @Property(name: 'lowPrice')
  double lowPrice = null;
  

  @Property(name: 'lastPrice')
  double lastPrice = null;
  

  @Property(name: 'lastPriceProtected')
  double lastPriceProtected = null;
  

  @Property(name: 'lastTickDirection')
  String lastTickDirection = null;
  

  @Property(name: 'lastChangePcnt')
  double lastChangePcnt = null;
  

  @Property(name: 'bidPrice')
  double bidPrice = null;
  

  @Property(name: 'midPrice')
  double midPrice = null;
  

  @Property(name: 'askPrice')
  double askPrice = null;
  

  @Property(name: 'impactBidPrice')
  double impactBidPrice = null;
  

  @Property(name: 'impactMidPrice')
  double impactMidPrice = null;
  

  @Property(name: 'impactAskPrice')
  double impactAskPrice = null;
  

  @Property(name: 'hasLiquidity')
  bool hasLiquidity = null;
  

  @Property(name: 'openInterest')
  Number openInterest = null;
  

  @Property(name: 'openValue')
  Number openValue = null;
  

  @Property(name: 'fairMethod')
  String fairMethod = null;
  

  @Property(name: 'fairBasisRate')
  double fairBasisRate = null;
  

  @Property(name: 'fairBasis')
  double fairBasis = null;
  

  @Property(name: 'fairPrice')
  double fairPrice = null;
  

  @Property(name: 'markMethod')
  String markMethod = null;
  

  @Property(name: 'markPrice')
  double markPrice = null;
  

  @Property(name: 'indicativeTaxRate')
  double indicativeTaxRate = null;
  

  @Property(name: 'indicativeSettlePrice')
  double indicativeSettlePrice = null;
  

  @Property(name: 'settledPrice')
  double settledPrice = null;
  

  @Property(name: 'timestamp')
  DateTime timestamp = null;
  
  Instrument();

  @override
  String toString()  {
    return 'Instrument[symbol=$symbol, rootSymbol=$rootSymbol, state=$state, typ=$typ, listing=$listing, front=$front, expiry=$expiry, settle=$settle, relistInterval=$relistInterval, inverseLeg=$inverseLeg, sellLeg=$sellLeg, buyLeg=$buyLeg, positionCurrency=$positionCurrency, underlying=$underlying, quoteCurrency=$quoteCurrency, underlyingSymbol=$underlyingSymbol, reference=$reference, referenceSymbol=$referenceSymbol, calcInterval=$calcInterval, publishInterval=$publishInterval, publishTime=$publishTime, maxOrderQty=$maxOrderQty, maxPrice=$maxPrice, lotSize=$lotSize, tickSize=$tickSize, multiplier=$multiplier, settlCurrency=$settlCurrency, underlyingToPositionMultiplier=$underlyingToPositionMultiplier, underlyingToSettleMultiplier=$underlyingToSettleMultiplier, quoteToSettleMultiplier=$quoteToSettleMultiplier, isQuanto=$isQuanto, isInverse=$isInverse, initMargin=$initMargin, maintMargin=$maintMargin, riskLimit=$riskLimit, riskStep=$riskStep, limit=$limit, capped=$capped, taxed=$taxed, deleverage=$deleverage, makerFee=$makerFee, takerFee=$takerFee, settlementFee=$settlementFee, insuranceFee=$insuranceFee, fundingBaseSymbol=$fundingBaseSymbol, fundingQuoteSymbol=$fundingQuoteSymbol, fundingPremiumSymbol=$fundingPremiumSymbol, fundingTimestamp=$fundingTimestamp, fundingInterval=$fundingInterval, fundingRate=$fundingRate, indicativeFundingRate=$indicativeFundingRate, rebalanceTimestamp=$rebalanceTimestamp, rebalanceInterval=$rebalanceInterval, openingTimestamp=$openingTimestamp, closingTimestamp=$closingTimestamp, sessionInterval=$sessionInterval, prevClosePrice=$prevClosePrice, limitDownPrice=$limitDownPrice, limitUpPrice=$limitUpPrice, bankruptLimitDownPrice=$bankruptLimitDownPrice, bankruptLimitUpPrice=$bankruptLimitUpPrice, prevTotalVolume=$prevTotalVolume, totalVolume=$totalVolume, volume=$volume, volume24h=$volume24h, prevTotalTurnover=$prevTotalTurnover, totalTurnover=$totalTurnover, turnover=$turnover, turnover24h=$turnover24h, prevPrice24h=$prevPrice24h, vwap=$vwap, highPrice=$highPrice, lowPrice=$lowPrice, lastPrice=$lastPrice, lastPriceProtected=$lastPriceProtected, lastTickDirection=$lastTickDirection, lastChangePcnt=$lastChangePcnt, bidPrice=$bidPrice, midPrice=$midPrice, askPrice=$askPrice, impactBidPrice=$impactBidPrice, impactMidPrice=$impactMidPrice, impactAskPrice=$impactAskPrice, hasLiquidity=$hasLiquidity, openInterest=$openInterest, openValue=$openValue, fairMethod=$fairMethod, fairBasisRate=$fairBasisRate, fairBasis=$fairBasis, fairPrice=$fairPrice, markMethod=$markMethod, markPrice=$markPrice, indicativeTaxRate=$indicativeTaxRate, indicativeSettlePrice=$indicativeSettlePrice, settledPrice=$settledPrice, timestamp=$timestamp, ]';
  }

}

